### PR TITLE
add support for v2 assignment overrides endpoint in fake server

### PR DESCRIPTION
**Pull request by samandmoore**
/domain @Betterment/test_track_core 
/no-platform

This PR adds support for the newly added v2 assignment override endpoint. https://github.com/Betterment/test_track/pull/145

I removed support for form url encoded POSTs for this new version of the endpoint. I did this partly because it simplified things and partly because I wasn't entirely sure how we'd wanna handle a POST request with a list of objects. I don't think it's all that valuable given the new request body format.

### CODE REVIEW

File: fakeserver/routes.go [hunk 1]
```diff
@@ -257,6 +261,35 @@ func postV1AssignmentOverride(r *http.Request) error {
...
+		requestBytes, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			return err
+		}
+		var assignmentMap map[string][]v1Assignment
+		err = json.Unmarshal(requestBytes, &assignmentMap)
+		if err != nil {
+			return err
+		}
+		assignments = assignmentMap["assignments"]
```
Comment (samandmoore) [inline fakeserver/routes.go:278]: keeping things simple. the request looks like this:

```json
{
  "assignments":[
    {"split_name": "allocation_selection_refactor_q1_2018_enabled", "variant": "true"}, 
    {"split_name":"add_preferred_muni_bond_locale_selection_to_dr_enabled", "variant":"false"}
  ]
}
```
so, i'm just unmarshaling that as a map of string to v1Assignment and then pulling the value for that one key

**Comment by smudge** [comment]
<< domain LGTM!

File: fakeserver/routes.go [hunk 2]
```diff
@@ -257,6 +261,35 @@ func postV1AssignmentOverride(r *http.Request) error {
...
+	default:
+		return fmt.Errorf("got unexpected content type %s", contentType)
+	}
+	storedAssignments, err := fakeassignments.Read()
+	for _, assignment := range assignments {
+		(*storedAssignments)[assignment.SplitName] = assignment.Variant
+	}
```
Comment (samandmoore) [inline fakeserver/routes.go:285]: this is the other new bit compared to the previous singular version. before this was a single assignment, now it's a loop ➰

**Comment by samandmoore** [comment]
nanda?

File: fakeserver/routes.go [hunk 3]
```diff
@@ -26,6 +26,11 @@ type v1Assignment struct {
 	Unsynced  bool   `json:"unsynced"`
 }
 
+// v2AssignmentOverrideRequestBody is the JSON input for the V2 assignment override endpoint
+type v2AssignmentOverrideRequestBody struct {
+	Assignments []v1Assignment `json:"assignments"`
+}
+
```
Comment (samandmoore) [inline fakeserver/routes.go:33]: ended up introducing this to help with deserialization

File: fakeserver/server_test.go [hunk 4]
```diff
@@ -150,3 +155,40 @@ func TestPersistAssignment(t *testing.T) {
 		require.Equal(t, "control", (*assignments)["test.test_experiment"])
 	})
 }
+
+func TestPersistAssignmentV2(t *testing.T) {
```
Comment (samandmoore) [inline fakeserver/server_test.go:159]: @smudge i snuck a test in here too since i noticed that there was one for the old way

**Pull request merged**